### ISSUE TITLE
Support `assert*` calls on Laravel test fakes (e.g. `Cache::assertNothingDispatched()`)

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#791]
## Feature request

Resolve `Facade::assert*()` calls against their matching `*Fake` class, so Laravel test suites stop emitting false-positive `UndefinedMagicMethod` on every assertion.

## Examples that currently fail

```php
Cache::fake();

$this->service->runJob();

Cache::assertNothingDispatched();                 // UndefinedMagicMethod
Event::assertDispatched(UserCreated::class);      // UndefinedMagicMethod
Queue::assertPushed(SendWelcomeEmail::class);     // UndefinedMagicMethod
Mail::assertSent(OrderShipped::class);            // UndefinedMagicMethod
Notification::assertSentTo($user, PasswordReset::class);
Storage::assertExists('uploads/avatar.jpg');
Http::assertSent(fn ($r) => $r->url() === '...');
Bus::assertDispatched(ProcessPayment::class);
Process::assertRan('ls -la');
```

All of these are valid Laravel test code. The `assert*` methods live on `\Illuminate\Support\Testing\Fakes\*Fake` classes, not on the real underlying services — so our current resolver, which walks the real classes, can't see them.

## How Larastan solves it

`larastan/src/Methods/FacadesMethodsExtension.php:75-104`:

```php
if (Str::startsWith($methodName, 'assert')) {
    $fakeFacadeClass = $this->getFake($facadeClass);

    if ($this->reflectionProvider->hasClass($fakeFacadeClass)) {
        $fakeReflection = $this->reflectionProvider->getClass($fakeFacadeClass);

        if ($fakeReflection->hasMethod($methodName)) {
            $this->cache[$key] = new StaticMethodReflection(
                $fakeReflection->getMethod($methodName, new OutOfClassScope()),
            );
            return true;
        }
    }
}

private function getFake(string $facade): string
{
    $shortClassName = substr($facade, strrpos($facade, '\\') + 1);
    return sprintf('\\Illuminate\\Support\\Testing\\Fakes\\%sFake', $shortClassName);
}
```

Naming-convention-only: `Cache` → `CacheFake`, `Event` → `EventFake`, etc. No runtime execution, no container probing.

## Proposed implementation

Add a fifth resolution path in `FacadeMethodHandler::resolveMethod()` (after path 4 `tryGetFacadeRootClass`, before the final `null` cache-and-return):

```php
if (\str_starts_with($methodNameLower, 'assert')) {
    $fakeClass = self::deriveFakeClass($facadeClass);
    if ($fakeClass !== null) {
        $resolved = self::lookupPublicMethod($codebase, $fakeClass, $methodNameLower);
        if ($resolved !== null) {
            return self::$methodCache[$key] = $resolved;
        }
    }
}
```

New private helper:

```php
private static function deriveFakeClass(string $facadeClass): ?string
{
    $shortName = \substr($facadeClass, \strrpos($facadeClass, '\\') + 1);
    if ($shortName === '') {
        return null;
    }
    $candidate = '\\Illuminate\\Support\\Testing\\Fakes\\' . $shortName . 'Fake';

    return \class_exists($candidate, true) ? \ltrim($candidate, '\\') : null;
}
```

The fake class needs to be scanned for `classlike_storage_provider->get()` to succeed. Easiest: queue it in `AppFacadeRegistrationHandler::registerHandlersForFacade()` when we register a facade — cheap and deterministic.

### Graceful degradation

Not every first-party facade has a matching `*Fake`:

- Has fake: `Cache`, `Event`, `Queue`, `Mail`, `Notification`, `Storage`, `Http`, `Bus`, `Process`.
- No fake: `View`, `Config`, `Lang`, `App`, `URL`, `Session` (and most others).

`class_exists()` returning false cleanly falls through to the final `null`, so `View::assertFoo()` still emits `UndefinedMagicMethod` — correct.

## Scope

Applies to any class extending `\Illuminate\Support\Facades\Facade` (first-party AND app-owned, both under #789's registration path). The gate is method-name-prefix, not facade-class identity, so no separate handler.

### Files to modify

- `src/Handlers/Facades/FacadeMethodHandler.php` — add path 5 + `deriveFakeClass()` helper.
- `src/Handlers/Facades/AppFacadeRegistrationHandler.php` — queue fake class for scanning at registration time.
- `tests/Type/tests/Facades/` — new PHPT.
- `tests/Unit/Handlers/Facades/FacadeMethodHandlerTest.php` — unit coverage for `deriveFakeClass`.

### Acceptance criteria

- PHPT: `Cache::assertNothingDispatched()` resolves — no `UndefinedMagicMethod`.
- PHPT: `Event::assertDispatched(Foo::class, 1)` resolves with correct param types.
- PHPT: `Storage::assertExists('path/to/file')` resolves.
- PHPT: `View::assertSomething()` still emits `UndefinedMagicMethod` (no `ViewFake` class exists).
- PHPT: `Cache::get('key')` still returns the real `get()` type — `assert*` path doesn't steal non-assert methods.
- Unit: `deriveFakeClass` edge cases (empty FQCN, FQCN with no separator, non-existent fake).
- `composer test` and `composer test:app` clean.

## Out-of-scope

- `partialMock`, `spy`, and other non-`assert*` test-facade methods — different naming, separate discussion.
- `Http::fake([...])` array-argument overloads — covered by existing `Http` stubs.

## Related

- #789 — app-owned facade resolution. This issue reuses the same resolver entry point but doesn't depend on that PR shipping first (though it's cleanest to land after #789 since the resolver file already exists).
- #790 — real-time facades. Orthogonal.

Estimated size: ~30-50 LoC + 4-6 PHPT scenarios + 2-3 unit tests. Low risk, high user value — every Laravel test suite uses these assertions.

**Comment by alies-dev** [comment]
After investigation, the premise here is incorrect for current Laravel: Laravel's first-party facades already declare `@method static void assertX(...)` for every assert method on their corresponding `*Fake`. Verified 1:1 coverage on Laravel 12.56 and 13.5 for `Bus`, `Event`, `Queue`, `Mail`, `Notification` (the only facades with a matching `*Fake` in `Illuminate\Support\Testing\Fakes\`).

Concretely, on master with the plugin installed:

```php
Bus::assertNothingDispatched();           // resolves via @method, no UndefinedMagicMethod
Event::assertDispatched(Foo::class);      // resolves via @method
Queue::assertPushed(Job::class);          // resolves via @method
Mail::assertSent(Mailable::class);        // resolves via @method
Notification::assertSentTo($u, N::class); // resolves via @method
```

Larastan needs the fake-class lookup because PHPStan handles facades differently. Psalm's pseudo-method resolution already covers `@method` tags natively, so this plugin doesn't need a parallel mechanism.

The examples in the original report that genuinely don't resolve (`Cache::assertNothingDispatched()`, `Storage::assertExists()`, `Http::assertSent()`, `Process::assertRan()`) are not solvable via the proposed naming convention either. Cache has no fake at all (the listed assertion doesn't exist on any Cache class). For Storage/Http/Process, `::fake()` swaps in `FilesystemAdapter` / `Http\Client\Factory` / `Process\Factory` — assertion methods live there, not on a `*Fake`-named class. Resolving those would require static-time knowledge of `::fake()` rebindings, which is out of scope for naming-convention resolution.

Closing as not planned. Reopening welcome if a concrete failing case turns up that isn't covered by the current `@method` declarations.
